client/src/components/Pagination.tsx

```diff
@@ -0,0 +1,39 @@
+export type PaginationProps = {
+  itemsPerPage: number;
+  totalItems: number | undefined;
+  currentPage: number;
+  paginate: (number: number) => void;
+};
+
+export default function Pagination({
+  itemsPerPage,
+  totalItems,
+  paginate,
+  currentPage,
+}: PaginationProps) {
+  const pageNumbers = [];
+
+  if (totalItems) {
+    for (let i = 1; i <= Math.ceil(totalItems / itemsPerPage); i++) {
+      pageNumbers.push(i);
+    }
+  }
+
+  return (
+    <nav className="mt-4 py-2 border-t border-b w-full max-w-152">
+      <ul className="flex justify-center gap-2">
+        {pageNumbers.map((number) => (
```

Comment:
Here, `number` is again confusing because of its meaning as a TS type. Would `page` or `pageNo` or similar be better?